src/router.ts

```diff
@@ -0,0 +1,121 @@
+import IResourceStore from './IResourceStore'
+import IResourceIdentifier from './IResourceIdentifier'
+import IRepresentation from './IRepresentation'
+import toFolderDescription from './toFolderDescription'
+import IResponse from './IResponse'
+import * as Stream from 'stream'
+import sha256 from './sha256'
+
+function stringToStream ( text: string ) : Stream {
+  const stream = new Stream.Readable()
+  stream._read = () => {}
+  stream.push(text)
+  stream.push(null)
+  return stream
+}
+
+function make412Response () {
```

Comment:
this is weird

Reply:
yeah, it's just a draft so far, lots of ad-hoc code like this in various place still!

Reply:
not ready for review yet, just pushed it here because Jackson asked me to look at the tooling